src/app/channels/channels.component.html

```diff
@@ -1,15 +1,15 @@
 <h1>Channels</h1>
 <mat-accordion>
-    <mat-expansion-panel *ngFor="let channel of channels">
+    <mat-expansion-panel *ngFor="let channel of channels" [id]="this.getChannelIdentifier(channel)" [expanded]="selectedChannel == this.getChannelIdentifier(channel)" (opened)="this.setChannelSelection(channel)">
```

Comment:
I think we should store the "Channel Identifier" as a variable on the channel instead of using a function in the component to get it.  Functions on components run a LOT and this seems like something that would be really easy to just define once and re-use.

Reply:
Thanks for the advice, I am also learning a lot from this.
I will open a PR soon with your suggestion

Reply:
if you wanna get an idea how much those functions run.. throw a console.log in one and watch it go crazy haha.  I still use them for things... just not when it can be calculated once and stored.. and only when the logic in the function is dead simple.

Reply:
Oh, I see. Yes, better to calculate it once.
PR got a bit bigger since I split up the models. Feel free to have a look and review https://github.com/springwolf/springwolf-ui/pull/24

To avoid the publish/subscribe confusion I opened another PR using your badges. Feel free to have a look as well - now with previews enabled